### ISSUE TITLE
Different running result in two machines

### ISSUE THREAD

**Issue by licstar** [dmlc/mxnet#1132]
I have two machines: A and B. However I got different results when running example/autoencoder.

I trained model on A, the last lines are:

> INFO:root:Iter:99000 metric:0.156347
> INFO:root:At Iteration [100000]: Swith to new learning rate 0.00000

> Training error: 0.149540036917
> Validation error: 0.177815541625

I copy the model file mnist_pt.arg to B, comment the training lines and only left load and evaluation lines. The result are as follows:

> Training error: 0.156354278326
> Validation error: 0.179336279631

The results are stable in each machine, no matter I use cpu or gpu.

I guess machine B generates the correct result, because 0.156347 and 0.149540036917 are quite different.

So, why the difference?.

----------------------
edit:

I run the demo on machine C, the result is the same as machine A.

**Comment by antinucleon** [comment]
Do you have dropout or other randomness in your network?

**Comment by licstar** [comment]
It seems the example contains dropout. https://github.com/dmlc/mxnet/blob/master/example/autoencoder/autoencoder.py

To my understand, dropout can generate same results in evaluation phase. In this experiment, I load pre-trained model from the same file, and the two machine generate different results.

**Comment by antinucleon** [comment]
Are you using is train in forward pass?

**Comment by licstar** [comment]
code in evaluation:
https://github.com/dmlc/mxnet/blob/bd84b2c71a9c469da37b87787db36a9d7998c768/example/autoencoder/model.py#L24

> exe.forward(is_train=False)

**Comment by licstar** [comment]
I checked the internal values in this example. I found the relative error for every single value is after the 7th significant digit. I think it's the limitation of single precision float point value. After the last step, calculating the mean of these values, the error was enlarged.

So, I think it's not belong to the issue of MxNet
